packages/node/src/Stream/Experimental/index.ts

```diff
@@ -0,0 +1,129 @@
+/**
+ * ets_tracing: off
+ */
+
+import * as C from "@effect-ts/core/Collections/Immutable/Chunk"
+import * as T from "@effect-ts/core/Effect"
+import * as S from "@effect-ts/core/Effect/Experimental/Stream"
+import * as Sink from "@effect-ts/core/Effect/Experimental/Stream/Sink"
+import * as M from "@effect-ts/core/Effect/Managed"
+import { pipe } from "@effect-ts/core/Function"
+import type * as stream from "stream"
+
+import * as Byte from "../../Byte"
+
+export class ReadableError {
+  readonly _tag = "ReadableError"
+  constructor(readonly error: Error) {}
+}
+
+/**
+ * Captures a Node `Readable`, converting it into a `Stream`,
+ *
+ * Note: your Readable should not have an encoding set in order to work with buffers,
+ * calling this with a Readable with an encoding set will `Die`.
+ */
+export function streamFromReadable(
+  r: () => stream.Readable
+): S.Stream<unknown, ReadableError, Byte.Byte> {
+  return pipe(
+    T.succeedWith(r),
+    T.tap((sr) =>
+      sr.readableEncoding != null
+        ? T.dieMessage(
+            `stream.Readable encoding set to ${sr.readableEncoding} cannot be used to produce Buffer`
+          )
+        : T.unit
+    ),
+    S.acquireReleaseWith((sr) =>
+      T.succeedWith(() => {
+        sr.destroy()
+      })
+    ),
+    S.chain((sr) =>
+      S.async<unknown, ReadableError, Byte.Byte>((emit) => {
+        sr.on("data", (data) => {
```

Comment:
We could use the 'readable' event for better performance. That could also make it possible to split bigger buffers into chunks of specific size. (DEFAULT_CHUNK_SIZE)
```typescript
.on('readable', () => {
  let buffer: Buffer | null = null;
  
  while ((buffer = sr.read(bufferSize)) !== null) {
    emit.chunk(Byte.chunk(data))
  }
})

Reply:
The only thing I worry about is the following piece of documentation about the `size` argument to `read()`:

> The optional size argument specifies a specific number of bytes to read. If size bytes are not available to be read, 
> null will be returned _unless_ the stream has ended, in which case all of the data remaining in the internal buffer
> will be returned.

I suppose we can use the `readable` event for now and see if we run into any issues. See https://github.com/Effect-TS/node/pull/382/commits/dd2eea38bde5623ff9d6a40039087d9d4094fc6f.